src/main/java/gyro/aws/elasticsearch/ElasticsearchDomainResource.java

```diff
@@ -0,0 +1,606 @@
+
+        if (model.elasticsearchClusterConfig() != null) {
+            ElasticsearchClusterConfiguration elasticsearchClusterConfiguration = newSubresource(
+                ElasticsearchClusterConfiguration.class);
+            elasticsearchClusterConfiguration.copyFrom(model.elasticsearchClusterConfig());
+            setClusterConfiguration(elasticsearchClusterConfiguration);
+        }
+
+        if (model.snapshotOptions() != null) {
+            ElasticsearchSnapshotOptions elasticsearchSnapshotOptions = newSubresource(ElasticsearchSnapshotOptions.class);
+            elasticsearchSnapshotOptions.copyFrom(model.snapshotOptions());
+            setSnapshotOptions(elasticsearchSnapshotOptions);
+        }
+
+        if (model.nodeToNodeEncryptionOptions() != null) {
+            ElasticsearchNodeToNodeEncryptionOptions elasticsearchNodeToNodeEncryptionOptions = newSubresource(
+                ElasticsearchNodeToNodeEncryptionOptions.class);
+            elasticsearchNodeToNodeEncryptionOptions.copyFrom(model.nodeToNodeEncryptionOptions());
+            setNodeToNodeEncryptionOptions(elasticsearchNodeToNodeEncryptionOptions);
+        }
+
+        if (model.domainEndpointOptions() != null) {
+            ElasticsearchDomainEndpointOptions elasticsearchDomainEndpointOptions = newSubresource(
+                ElasticsearchDomainEndpointOptions.class);
+            elasticsearchDomainEndpointOptions.copyFrom(model.domainEndpointOptions());
+            setDomainEndpointOptions(elasticsearchDomainEndpointOptions);
+        }
+
+        if (model.encryptionAtRestOptions() != null) {
+            ElasticsearchEncryptionAtRestOptions elasticsearchEncryptionAtRestOptions = newSubresource(
+                ElasticsearchEncryptionAtRestOptions.class);
+            elasticsearchEncryptionAtRestOptions.copyFrom(model.encryptionAtRestOptions());
+            setEncryptionAtRestOptions(elasticsearchEncryptionAtRestOptions);
+        }
+
+        if (model.vpcOptions() != null) {
+            ElasticsearchVpcOptions elasticsearchVpcOptions = newSubresource(ElasticsearchVpcOptions.class);
+            elasticsearchVpcOptions.copyFrom(model.vpcOptions());
+            setVpcOptions(elasticsearchVpcOptions);
+        }
+
+        if (model.advancedSecurityOptions() != null) {
+            ElasticsearchAdvancedSecurityOptions elasticsearchAdvancedSecurityOptions = newSubresource(
+                ElasticsearchAdvancedSecurityOptions.class);
+            elasticsearchAdvancedSecurityOptions.copyFrom(model.advancedSecurityOptions());
+            setAdvancedSecurityOptions(elasticsearchAdvancedSecurityOptions);
+        }
+
+        if (refreshTags) {
+            ElasticsearchClient client = createClient(ElasticsearchClient.class);
+
+            getTags().clear();
+
+            client.listTags(r -> r.arn(getArn())).tagList().forEach(t -> getTags().put(t.key(), t.value()));
+        }
+    }
+
+    private void addAllTags(ElasticsearchClient client) {
```

Comment:
Unless there is a method that adds limited tags, no reason to name the method `addAllTags`
```suggestion
    private void addTags(ElasticsearchClient client) {
```